clients/src/main/java/org/apache/kafka/common/config/internals/ConfigProviderUtils.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.config.internals;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ConfigProviderUtils {
+
+    /**
+     * Reads the given {@code configValue} and creates a list of paths.
+     * @param configValue allowed.paths config value which is a string containing comma separated list of paths
+     * @return List of paths or null if the {@code configValue} is null or empty string.
+     */
+    public static List<Path> configureAllowedPaths(String configValue) {
+        if (configValue != null && !configValue.isEmpty()) {
+            List<Path> allowedPaths = new ArrayList<>();
+            Arrays.stream(configValue.split(",")).forEach(b -> allowedPaths.add(Paths.get(b).normalize()));
```

Comment:
I wonder if we should also enforce each path is absolute too. This would avoid users configuring the allowed paths with `../path` for example. WDYT?

Reply:
You still need to normalize them to remove `.` and `..` in the middle of the path (for example `/tmp/../path`) but I think it would be preferable to ensure allowed paths are absolute. Should we also ensure they exist or are there use cases where the paths might be created later?

Reply:
I think it makes sense to ensure that paths are absolute and exist. I have taken the suggestion in the latest commit.